WireAPI/Sources/WireAPI/APIs/ConversationsAPI/ConversationsAPIV0.swift

```diff
@@ -96,3 +124,115 @@ private struct PaginatedConversationIDsV0: Decodable, ToAPIModelConvertible {
+
+    func toAPIModel() -> ConversationList {
+        ConversationList(
+            found: found.map { $0.toAPIModel() },
+            notFound: notFound,
+            failed: failed
+        )
+    }
+}
+
+// MARK: -
+
+struct ConversationV0: Decodable, ToAPIModelConvertible {
+    enum CodingKeys: String, CodingKey {
+        case access
+        case accessRole = "access_role"
+        case accessRoleV2 = "access_role_v2"
+        case creator
+        case epoch
+        case id
+        case lastEvent = "last_event"
+        case lastEventTime = "last_event_time"
+        case members
+        case messageProtocol = "protocol"
+        case messageTimer = "message_timer"
+        case mlsGroupID = "group_id"
+        case name
+        case qualifiedID = "qualified_id"
+        case readReceiptMode = "receipt_mode"
+        case teamID = "team"
+        case type
+    }
+
+    var access: [String]?
+    var accessRoles: [String]?
+    var creator: UUID?
+    var epoch: UInt?
+    var id: UUID?
+    var lastEvent: String?
+    var lastEventTime: String?
+    var legacyAccessRole: String?
+    var members: QualifiedConversationMembers?
+    var messageProtocol: String?
+    var messageTimer: TimeInterval?
+    var mlsGroupID: String?
+    var name: String?
+    var qualifiedID: QualifiedID?
+    var readReceiptMode: Int?
+    var teamID: UUID?
+    var type: Int?
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        access = try container.decodeIfPresent([String].self, forKey: .access)
+        creator = try container.decodeIfPresent(UUID.self, forKey: .creator)
+        epoch = try container.decodeIfPresent(UInt.self, forKey: .epoch)
+        id = try container.decodeIfPresent(UUID.self, forKey: .id)
+        lastEvent = try container.decodeIfPresent(String.self, forKey: .lastEvent)
+        lastEventTime = try container.decodeIfPresent(String.self, forKey: .lastEventTime)
+        members = try container.decodeIfPresent(QualifiedConversationMembers.self, forKey: .members)
+        messageProtocol = try container.decodeIfPresent(String.self, forKey: .messageProtocol)
+        messageTimer = try container.decodeIfPresent(TimeInterval.self, forKey: .messageTimer)
+        mlsGroupID = try container.decodeIfPresent(String.self, forKey: .mlsGroupID)
+        name = try container.decodeIfPresent(String.self, forKey: .name)
+        qualifiedID = try container.decodeIfPresent(QualifiedID.self, forKey: .qualifiedID)
+        readReceiptMode = try container.decodeIfPresent(Int.self, forKey: .readReceiptMode)
+        teamID = try container.decodeIfPresent(UUID.self, forKey: .teamID)
+        type = try container.decodeIfPresent(Int.self, forKey: .type)
+
+        // parsing for v0
+        legacyAccessRole = try container.decodeIfPresent(String.self, forKey: .accessRole)
+        accessRoles = try container.decodeIfPresent([String].self, forKey: .accessRoleV2)
```

Comment:
Is this custom decoding necessary? If we rely on the generated conformance, then we'd still understand which of the `accessRole` and `accessRoleV2` fields get mapped to `legacyAccessRole` and `acessRoles` when converting to the api model.

Reply:
right, good catch https://github.com/wireapp/wire-ios/commit/03fd6767da32b530f841978ffcafbdfa58dcaf9b